tfe/resource_tfe_variable_set_test.go

```diff
@@ -0,0 +1,264 @@
+			ctx,
+			rs.Primary.ID,
+			&tfe.VariableSetReadOptions{Include: &[]tfe.VariableSetIncludeOpt{tfe.VariableSetWorkspaces}},
+		)
+		if err != nil {
+			return err
+		}
+
+		*variableSet = *vs
+
+		return nil
+	}
+}
+
+func testAccCheckTFEVariableSetAttributes(
+	variableSet *tfe.VariableSet) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		if variableSet.Name != "variable_set_test" {
+			return fmt.Errorf("Bad name: %s", variableSet.Name)
+		}
+		if variableSet.Description != "a test variable set" {
+			return fmt.Errorf("Bad description: %s", variableSet.Description)
+		}
+		if variableSet.Global != false {
+			return fmt.Errorf("Bad global: %t", variableSet.Global)
+		}
+
+		return nil
+	}
+}
+
+func testAccCheckTFEVariableSetAttributesUpdate(
+	variableSet *tfe.VariableSet) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		if variableSet.Name != "variable_set_test_updated" {
+			return fmt.Errorf("Bad name: %s", variableSet.Name)
+		}
+		if variableSet.Description != "another description" {
+			return fmt.Errorf("Bad description: %s", variableSet.Description)
+		}
+		if variableSet.Global != true {
+			return fmt.Errorf("Bad global: %t", variableSet.Global)
+		}
+
+		return nil
+	}
+}
+
+func testAccCheckTFEVariableSetApplication(variableSet *tfe.VariableSet) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		if len(variableSet.Workspaces) != 1 {
+			return fmt.Errorf("Bad workspace apply: %v", variableSet.Workspaces)
+		}
+
+		return nil
+	}
+}
+
+func testAccCheckTFEVariableSetApplicationUpdate(variableSet *tfe.VariableSet) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		if len(variableSet.Workspaces) != 0 {
+			return fmt.Errorf("Bad workspace apply: %v", variableSet.Workspaces)
+		}
+
+		return nil
+	}
+}
+
+func testAccCheckTFEVariableSetDestroy(s *terraform.State) error {
+	tfeClient := testAccProvider.Meta().(*tfe.Client)
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "tfe_variable_set" {
+			continue
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No instance ID is set")
+		}
+
+		_, err := tfeClient.VariableSets.Read(ctx, rs.Primary.ID, nil)
+		if err == nil {
+			return fmt.Errorf("Variable Set %s still exists", rs.Primary.ID)
+		}
+	}
+
+	return nil
+}
+
+func testAccTFEVariableSet_basic(rInt int) string {
+	return fmt.Sprintf(`
+resource "tfe_organization" "foobar" {
+  name = "tst-terraform-%d"
+	email = "admin@company.com"
+}
+
+resource "tfe_workspace" "foobar" {
+  name = "foobar"
+	organization = tfe_organization.foobar.id
+}
+
+resource "tfe_variable_set" "foobar" {
+  name         = "variable_set_test"
+	description  = "a test variable set"
+	global       = false
+	organization = tfe_organization.foobar.id
+}
+
+resource "tfe_variable_set" "applied" {
+  name         = "variable_set_applied"
+	description  = "a test variable set"
+	workspaces   = [tfe_workspace.foobar.id]
```

Comment:
Isn't workspaces a computed attribute? We should omit computed attributes from our configuration, what do you think?